BookkeeperAPI/Controllers/ExpenseController.cs

```diff
@@ -3,24 +3,155 @@
     #region usings
     using BookkeeperAPI.Data;
     using BookkeeperAPI.Entity;
+    using BookkeeperAPI.Model;
     using Microsoft.AspNetCore.Mvc;
+    using Microsoft.EntityFrameworkCore;
+    using BookkeeperAPI.Constants;
+    using BookkeeperAPI.Utility;
     #endregion
 
     [ApiController]
-    [Route("/api/v1/user/{userId}/expenses")]
     [Produces("application/json")]
     public class ExpenseController : ControllerBase
     {
-        private BookkeeperContext _context;
-        public ExpenseController(BookkeeperContext context) 
+        private readonly BookkeeperContext _context;
+        public ExpenseController(BookkeeperContext context)
         {
             _context = context;
         }
 
-        [HttpGet]
-        public ActionResult<List<Expense>> GetExpense(Guid userId)
+        [HttpGet("/api/me/expenses")]
+        public async Task<ActionResult<List<ExpenseView>>> GetExpense(Guid userId)
         {
-            return Ok(_context.Expenses.Where(x => x.User.Id.Equals(userId)));
+            if(userId == Guid.Empty)
+            {
+                return BadRequest();
+            }
+
+            return Ok(
+                await _context.Expenses
+                .Where(x => x.UserId.Equals(userId))
+                .Select(x => new ExpenseView()
+                {
+                    Id = x.Id,
+                    Name = x.Name,
+                    Amount = x.Amount,
+                    Category = x.Category.ToString(),
+                    Date = x.Date,
+                }).ToListAsync());
+        }
+
+        [HttpGet("/api/expenses/{expenseId}")]
+        public async Task<ActionResult<ExpenseView>> GetExpenseById(Guid userId, Guid expenseId)
```

Comment:
If we are fetching expense by Id, there is no requirement for userId... unless one expenseId can belong to multiple users.